BridgeAppSDK/SBAScheduledActivityManager.swift

```diff
@@ -382,13 +397,54 @@ public class SBAScheduledActivityManager: NSObject, SBASharedInfoController, ORK
     public func sendUpdatedScheduledActivities(scheduledActivities: [SBBScheduledActivity]) {
         SBAUserBridgeManager.updateScheduledActivities(scheduledActivities)
     }
-    
-    public func archiveResults(schedule: SBBScheduledActivity, taskViewController: ORKTaskViewController) {
+        
+    public func archiveResults(schedule: SBBScheduledActivity, taskViewController: ORKTaskViewController) -> [SBADataArchive]? {
         
         let results = activityResultsForSchedule(schedule, taskViewController: taskViewController)
         print(results)
         
-        // TODO: implement syoung 04/27/2016 Stubbed out accessor
+        var archives = [SBADataArchive]()
+        for activityResult in results {
+            if let activityResultResults = activityResult.results as? [ORKStepResult] {
+                let archive = SBADataArchive.init(reference: activityResult.schemaIdentifier, schemaRevision: activityResult.schemaRevision)
+                for stepResult in activityResultResults {
+                    if let stepResultResults = stepResult.results {
+                        for result in stepResultResults {
+                            
+                            //Update date if needed
+                            if (result.startDate == nil) {
+                                result.startDate = stepResult.startDate;
+                                result.endDate = stepResult.endDate;
+                            }
+                            
+                            guard let archiveableResult = result.bridgeData(stepResult.identifier) else {
+                                assertionFailure("Something went wrong getting result to archive from result \(result.identifier) of step \(stepResult.identifier) of activity result \(activityResult.identifier)")
+                                archive.removeArchive()
+                                return nil
+                            }
+                            
+                            switch archiveableResult.resultType {
```

Comment:
This feels like it belongs on ArchiveResult instead. Is it here b/c you are considering grouping the results?